laurent22/joplin
# Mobile: Fixes #6663: Fix watchInjectedJs and support multiple output bundles

# Summary 
 * Previously, `watchInjectedJs` only watched `CodeMirror.ts` for changes. It now uses `rollup`'s JS API to watch all `.ts` files used to build the bundle.
     * `rollup.watch` should do an incremental build, so should be faster than the current complete rebuild.
 * As a result of switching to `rollup`'s JS API (away from the command line API), this allows more bundles to be added in the future.
     * The work-in-progress image editor will be using a separate bundle.

Fixes #6663.

# Testing plan
 * Run `yarn watchInjectedJs`
 * Run the mobile app on a device/simulator/emulator
 * Make a noticeable change to `CodeMirror.ts` (perhaps change the initialText of the document).
 * Check whether the bundle re-built
 * Check whether the re-built bundle can be properly loaded in the emulator
